django/db/models/functions.py

```diff
@@ -41,17 +41,17 @@ class ConcatPair(Func):
     def __init__(self, left, right, **extra):
         super(ConcatPair, self).__init__(left, right, **extra)
 
-    def as_sqlite(self, compiler, connection):
+    def as_sqlite(self, compiler, connection, **kwargs):
         coalesced = self.coalesce()
         coalesced.arg_joiner = ' || '
         coalesced.template = '%(expressions)s'
-        return super(ConcatPair, coalesced).as_sql(compiler, connection)
+        return super(ConcatPair, coalesced).as_sql(compiler, connection, **kwargs)
 
-    def as_mysql(self, compiler, connection):
+    def as_mysql(self, compiler, connection, **kwargs):
         # Use CONCAT_WS with an empty separator so that NULLs are ignored.
         self.function = 'CONCAT_WS'
         self.template = "%(function)s('', %(expressions)s)"
-        return super(ConcatPair, self).as_sql(compiler, connection)
+        return super(ConcatPair, self).as_sql(compiler, connection, **kwargs)
```

Comment:
Remove `self.function = 'CONCAT_WS' and the following line which mutates `self.template` and instead:

```
return super(ConcatPair, self).as_sql(compiler, connection, function='CONCAT_WS', template="%(function)s('' %(expression)s)"
```